core/templates/dev/head/pages/library/search-bar/search-bar.directive.js

```diff
@@ -0,0 +1,255 @@
+angular.module('searchBarModule').directive('searchBar', [
+    'UrlInterpolationService', function(UrlInterpolationService) {
+      return {
+        restrict: 'E',
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/library/search-bar/search-bar.template.html'),
```

Comment:
@seanlip This is the erring line.

Reply:
Hi @YashJipkate, I took a quick look. Take a look at your server logs (in your terminal) after you start up a local devserver and go to the library page. You'll see something like this:

```
WARNING  2019-03-19 08:40:19,841 base.py:453] Invalid URL requested: http://localhost:8181/templates/dev/head/pages/library/search-bar/search-bar.template.html
```

I think there's something wrong with the URL. I'm not sure whether it's the hyphens or the double-dots or the path root, but you can compare this with a URL that works and see what the difference is. (A debugging tip: you can pinpoint the specific error by changing an existing "known-to-be-good" file to have hyphens or double-dots etc., and see what causes that file URL to break.)

Hope this helps, please let us know what you find. Thanks!

Reply:
Hi @seanlip Thanks for taking a look into this.
Actually this was exactly the error I was referring to.
Also as you said, the first thing I tried was to change the template name to something which does not contain hyphen, dot to something like 'search_bar.html'. I even changed the directory name to 'search_bar'. But the same error popped up though with the changed file and directory name. 
Am I missing something here?

Reply:
Try it the other way round: start with another URL that you know works, and try to change it to look like your new one, and see where it breaks. Also, what is the usual request that is made to the backend when a (working) template URL is fetched -- does it start with /templates?

Reply:
I have tried using template directly instead of the URL and it works fine that way. Will try using some other Url too. 
Yeah it starts with /templates. You can see it in the UrlInterpolationService.getDirectiveTemplate that is used in rendering the template elsewhere, it appends /templates/dev/head to the url provided.

Reply:
Here's another thought -- take another look at the routing rules in app.yaml. Does your new filename conform to these rules?

Reply:
Oh thanks @seanlip it worked!!! :smile: Wasn't aware of the app.yaml part. It was the one blocking the rendering!
Thanks!